src/graph_notebook/configuration/generate_config.py

```diff
@@ -67,14 +67,20 @@ def __init__(self, host: str, port: int,
                  auth_mode: AuthModeEnum = DEFAULT_AUTH_MODE,
                  load_from_s3_arn='', ssl: bool = True, aws_region: str = DEFAULT_REGION,
                  proxy_host: str = '', proxy_port: int = DEFAULT_PORT,
-                 sparql_section: SparqlSection = None, gremlin_section: GremlinSection = None):
+                 sparql_section: SparqlSection = None, gremlin_section: GremlinSection = None,
+                 neptune_hosts: list = NEPTUNE_CONFIG_HOST_IDENTIFIERS):
         self.host = host
         self.port = port
         self.ssl = ssl
         self.proxy_host = proxy_host
         self.proxy_port = proxy_port
         self.sparql = sparql_section if sparql_section is not None else SparqlSection()
-        if "amazonaws.com" in self.host or "amazonaws.com" in self.proxy_host:
+        is_neptune_host = False
+        for host_snippet in neptune_hosts:
+            if host_snippet in self.host or host_snippet in self.proxy_host:
+                is_neptune_host = True
+                break
+        if is_neptune_host:
```

Comment:
I can see similar code in multiple places. Can we move this to a util method?

Reply:
Will do